test/logflare/sql_test.exs

```diff
@@ -487,6 +487,112 @@ defmodule Logflare.SqlTest do
     end
   end
 
+  describe "contains_cte?/2" do
+    test "returns true for queries with CTEs" do
+      query_with_cte = """
+      WITH users_summary AS (
+        SELECT user_id, COUNT(*) as total_events
+        FROM events
+        GROUP BY user_id
+      )
+      SELECT * FROM users_summary WHERE total_events > 10
+      """
+
+      assert Sql.contains_cte?(query_with_cte)
```

Comment:
I noticed this function is not used anywhere but I already had the tests. I am not sure if I should just delete it though :thinking: